src/server.c

```diff
@@ -2501,7 +2536,7 @@ void resetServerStats(void) {
     atomicSet(server.stat_total_writes_processed, 0);
     for (j = 0; j < STATS_METRIC_COUNT; j++) {
         server.inst_metric[j].idx = 0;
-        server.inst_metric[j].last_sample_time = mstime();
+        server.inst_metric[j].last_sample_time = j < STATS_METRIC_RATE_COUNT ? mstime() : 0;
```

Comment:
it is silly that this `mstime()` is called inside the loop (can be sampled once before it), but also, i don't feel comfortable with `STATS_METRIC_RATE_COUNT` (it's declaration seems odd, there are two constants with the value of `6`), and the coupling between how it's initialized and how it is used, is risky.
i'd rather all the initialization be the same, and also the extraction (getInstantaneousMetric) the same, and the only difference is if trackInstantaneousAvgMetric or trackInstantaneousRateMetric is called.

in that case, how about setting the initial last_sample_time to be 0 in both cases, and just make sure to skip the first time trackInstantaneousRateMetric is called (only do the init and exit right after)

what do you think?

Reply:
good idea